### PR TITLE
Issue 344: Add PLAN_ONLY parameter to PlanOnlyPlugin

### PR DESCRIPTION
**Pull request by duckpuppy**
* Issue #344 
* Create `PLAN_ONLY` parameter to control skipping `terraform apply`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@
 * [Issue #335](https://github.com/manheim/terraform-pipeline/issues/335) Testing: Rename DummyJenkinsfile to MockWorkflowScript to better distinguish Jenkinsfile references.
 * [Issue #271](https://github.com/manheim/terraform-pipeline/issues/271) Testing: Cleanup Test resets for any static state.  New Resettable and ResetStaticStateExtension available.
 * [Issue #332](https://github.com/manheim/terraform-pipeline/issues/332) Testing: Junit 4 to 5.7, upgrade hamcrest from 1.3 to 2.2, remove junit-hierarchicalcontextrunner dependency
+* [Issue #344](https://github.com/manheim/terraform-pipeline/issues/344) Add PLAN_ONLY parameter to PlanOnlyPlugin
+  * **BREAKING CHANGE** This change is a breaking change. Prior to this update, applying the PlanOnlyPlugin would restrict the pipeline to only running `terraform plan`. This update changes behavior to simply providing a `PLAN_ONLY` boolean parameter that can be set to restrict the build behavior. It defaults to `false`.
```
Comment (kmanning) [inline CHANGELOG.md:13]: v5.15 was published since this PR was started.  Can you modify the CHANGELOG accordingly?

File: docs/PlanOnlyPlugin.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ## [PlanOnlyPlugin](../src/PlanOnlyPlugin.groovy)
 
-Enable this plugin to change pipeline functionality to `terraform plan`.
+Enable this plugin to add a parameter to the build which will restrict pipeline functionality to `terraform plan` only.
```
Comment (kmanning) [inline docs/PlanOnlyPlugin.md:3]: This reads much better - thanks.

**Comment by kmanning** [comment]
Overall, looks great.  See CHANGELOG change above.

File: CHANGELOG.md (hunk 1 above)
Comment (duckpuppy) [reply CHANGELOG.md]: Oops, rebase error on my part not resolving that properly. Fixed in new commit.

**Comment by kmanning** [comment]
The updates look good, thanks!

**Pull request merged**